src/pmpo_fortran.f90

```diff
@@ -453,29 +449,32 @@ subroutine polympo_getMeshVtxRotLat(mpMesh, nVertices, latitude) &
   !> @param mpmesh(in/out) MPMesh object
   !> @param nComps(in) number of components, should always be 2
   !> @param nVertices(in) numVertices
-  !> @param array(in) input vertices velocity 2D array (2,numVtx)
+  !> @param uVel(in) vertices u-component velocity 1D array (numVtx)
+  !> @param vVel(in) vertices v-component velocity 1D array (numVtx)
   !---------------------------------------------------------------------------
-  subroutine polympo_setMeshVel(mpMesh, nComps, nVertices, array) &
+  subroutine polympo_setMeshVel(mpMesh, nComps, nVertices, uVel, vVel) &
              bind(C, NAME='polympo_setMeshVel_f')
     use :: iso_c_binding
     type(c_ptr), value :: mpMesh
     integer(c_int), value :: nComps, nVertices
-    type(c_ptr), intent(in), value :: array
+    type(c_ptr), intent(in), value :: uVel, vVel
   end subroutine
   !---------------------------------------------------------------------------
   !> @brief get the vertices velocity from polyMPO
   !> @param mpmesh(in/out) MPMesh object
   !> @param nComps(in) number of components, should always be 2
   !> @param nVertices(in) numVertices
-  !> @param array(in/out) output vertices velocity
-  !>        2D array (2,numVtx), allocated by user
+  !> @param uVel(in/out) output vertices u-component velocity
+  !>        1D array (numVtx), allocated by user
+  !> @param vVel(in/out) output vertices v-component velocity
+  !>        1D array (numVtx), allocated by user
   !---------------------------------------------------------------------------
-  subroutine polympo_getMeshVel(mpMesh, nComps, nVertices, array) &
+  subroutine polympo_getMeshVel(mpMesh, nComps, nVertices, uVel, vVel) &
```

Comment:
we do not need 'nComps' (see polympo_getMeshVtxCoords)

Reply:
removed https://github.com/SCOREC/polyMPO/commit/8bec689310aa9db8a2ecd8673a002858a0506d28